### PR TITLE
Move run tests to fastlane

### PR DESCRIPTION
**Pull request by WezSieTato**

### CODE REVIEW

File: fastlane/Fastfile-deploy [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  Steps:
+   * Create compiled framework
+   * Bump podspec version
+   * Commit framework and podspec changes as MenigaBuildsson
+   * Create version tag
+   * Push commit and tag to remote
+   * Push podspec to Trunk specs
+   * Merge branch into master
+  Options:  
+  - gh_user - Username fotr github account
```
Comment (arturdm) [inline fastlane/Fastfile-deploy]: typo - `fotr` -> `github username`?

File: fastlane/Fastfile-deploy [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+   * Create compiled framework
+   * Bump podspec version
+   * Commit framework and podspec changes as MenigaBuildsson
+   * Create version tag
+   * Push commit and tag to remote
+   * Push podspec to Trunk specs
+   * Merge branch into master
+  Options:  
+  - gh_user - Username fotr github account 
+  - gh_user_email - Email that be use to create commit
```
Comment (arturdm) [inline fastlane/Fastfile-deploy]: `Commit email address`?

File: fastlane/Fastfile-deploy [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  - gh_token - Access token for gh_user_name github account 
+  )
+  lane :ci_deploy do |options|
+    branch_name = git_branch
+    if !branch_name.match("release/*")
+      raise "ci_deploy should be run only from release branch!".red
+    end
+
+    if !options[:gh_user]
+      raise "No username specified!".red
```
Comment (arturdm) [inline fastlane/Fastfile-deploy]: `gh_user not specified`? I wonder if the message `No username specified` is readable enough.

File: fastlane/Fastfile-deploy [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    end
+    gh_user = options[:gh_user]
+
+    if !options[:gh_user_email]
+      raise "No email specified!".red
```
Comment (arturdm) [inline fastlane/Fastfile-deploy]: `gh_user_email not specified`? (same as with `gh_user`)

File: fastlane/Fastfile-deploy [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    end
+    gh_user_email = options[:gh_user_email]
+
+    if !options[:gh_token]
+    raise "No Github token specified!".red
```
Comment (arturdm) [inline fastlane/Fastfile-deploy]: formatting + the message does not point to which variable was not provided, or does it show it in some way?

File: fastlane/Fastfile-deploy [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    gh_token = options[:gh_token]
+
+    version = branch_name.partition("release/")[2]
+
+    build_framework
+    update_podspec_version(
+      version: version
+    )
+
+    gh_user = "MenigaBuildsson"
```
Comment (arturdm) [inline fastlane/Fastfile-deploy]: Why do we validate the options if we override it later on?

File: fastlane/Fastfile-deploy [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    set_git_config_user(
+      name: gh_user,
+      email: "mobile.dev@meniga.com"
```
Comment (arturdm) [inline fastlane/Fastfile-deploy]: Maybe we should extract those constants to the top of the file?

File: fastlane/Fastfile-deploy (hunk 6 above)
Comment (WezSieTato) [reply fastlane/Fastfile-deploy]: I want to have two lanes, one for CI with overrides options and one for manually deployment if something goes wrong with CI. With second case I as user of lane want to have proper messages if forgot about something.

File: fastlane/Fastfile-deploy (hunk 7 above)
Comment (WezSieTato) [reply fastlane/Fastfile-deploy]: What do you think about top of the lane? I'm prefer declaring constants as near place where is used and these constants are used only in this lane.

File: fastlane/Fastfile-deploy (hunk 6 above)
Comment (WezSieTato) [reply fastlane/Fastfile-deploy]: Ok, now I see what did you mean. I will remove overriding variables...

File: fastlane/Fastfile-deploy (hunk 7 above)
Comment (WezSieTato) [reply fastlane/Fastfile-deploy]: I will remove these constants, it should be provided with options.

File: fastlane/Fastfile-build [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+  
+    desc "Build framework under path Meniga/compiled"
+    lane :build_framework do
+      xcodebuild(
+        scheme: "Framework",
+        project: "Meniga/Meniga-ios-sdk.xcodeproj"
+      )
+    end
+  
+  end
```
Comment (arturdm) [inline fastlane/Fastfile-build]: no newline :)

**Pull request merged**